src/BinSkim.Driver/MultithreadedAnalyzeCommand.cs

```diff
@@ -71,13 +71,40 @@ protected override void InitializeConfiguration(AnalyzeOptions options, BinaryAn
 
         internal static string ReturnCommonPathRootFromTargetSpecifiersIfOneExists(IEnumerable<string> targetFileSpecifiers)
         {
+            var fileSpecifierDirectories = new SortedSet<string>(targetFileSpecifiers.Select(s => (Path.GetDirectoryName(Path.GetFullPath(s)) + @"\")),
+                                                                 StringComparer.OrdinalIgnoreCase);
 
-            var fileSpecifierDirectories = new HashSet<string>(targetFileSpecifiers.Select(s => Path.GetDirectoryName(Path.GetFullPath(s)) + @"\"),
-                                                               StringComparer.OrdinalIgnoreCase);
+            string smallestPath = fileSpecifierDirectories.First();
+            fileSpecifierDirectories.Remove(smallestPath);
 
-            return fileSpecifierDirectories.Count == 1
-                ? fileSpecifierDirectories.First()
-                : string.Empty;
+            for (int i = 0; i < smallestPath.Length; i++)
+            {
+                foreach (string specifier in fileSpecifierDirectories)
+                {
+                    if (char.ToLowerInvariant(smallestPath[i]) != char.ToLowerInvariant(specifier[i]))
+                    {
+                        smallestPath = smallestPath.Substring(0, i);
+                        break;
+                    }
+                }
+            }
+
+            if (smallestPath.Length == 0)
+            {
+                return string.Empty;
+            }
+
+            if (smallestPath[^1] != '\\')
+            {
+                // In this case, smallestPath is equal to 'c:\path1\partial-path' (this is an incomplete path).
+                // Once we execute, our smallestPath will be transformed into 'c:\path1\'.
+                string[] parts = smallestPath.Split('\\');
+                smallestPath = string.Join(@"\", parts[..^1]) + @"\";
+            }
+
+            return smallestPath.EndsWith(@"\\")
```

Comment:
>@"\\" [](http://example.com/codeflow?start=41&length=5)

when we have a path that is:
C:\*.dll, we are removing *.dll and adding a new \, becoming C:\\

So, this code is intended to remove the double slash at the end